src/pelicanfs/core.py

```diff
@@ -74,6 +84,30 @@ def __init__ (
         self._du = self.httpFileSystem._du
         self._info = self.httpFileSystem._info
 
+    def _discover_federation_metadata(self, discUrl):
+        """
+        Returns the json response from a GET call to the metadata discovery url of the federation
+        """
+        # Parse the url for federation discovery
+        discoveryUrl = urllib.parse.urlparse(discUrl)
+
+        discoveryUrl = discoveryUrl._replace(scheme='https')
+        updatedPath = discoveryUrl.path + "/.well-known/pelican-configuration"
+        discoveryUrl = discoveryUrl._replace(path=updatedPath)
+
+        resp = requests.get(discoveryUrl.geturl())
```

Comment:
As above - let's use the `aiohttp` client throughout.  Make this function async but, since `__init__` methods cannot be async, make it blocking at the current call site.